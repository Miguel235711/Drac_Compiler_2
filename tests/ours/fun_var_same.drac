var x;

fun(y){
    x = 3;
}

main(){
    main2 = 3;
    main2();
}
var main;
main(){ -- error, should fail because of redefinition
    var main2,main2; -- 2Â° main2 should fail because of redefinition
}
var main2;
main2(){
    var main2;
}
var main2;-- error, should fail because of redefinition