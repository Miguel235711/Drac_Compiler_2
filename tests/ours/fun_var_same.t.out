.
├── .\tests\ours\fun_var_same.drac
└── <program>
    └── <def-list>
        ├── <def-list>
        │   ├── <def-list>
        │   │   ├── <def-list>
        │   │   │   ├── <def-list>
        │   │   │   │   ├── <def-list>
        │   │   │   │   │   ├── <def-list>
        │   │   │   │   │   │   ├── <def-list>
        │   │   │   │   │   │   │   ├── <def-list>
        │   │   │   │   │   │   │   └── <def>
        │   │   │   │   │   │   │       └── <var-def>
        │   │   │   │   │   │   │           ├── var
        │   │   │   │   │   │   │           ├── <var-list>
        │   │   │   │   │   │   │           │   └── <id-list>
        │   │   │   │   │   │   │           │       ├── <id> -> table index: 0
        │   │   │   │   │   │   │           │       └── <id-list-cont>
        │   │   │   │   │   │   │           └── ;
        │   │   │   │   │   │   └── <def>
        │   │   │   │   │   │       └── <fun-def>
        │   │   │   │   │   │           ├── <id> -> table index: 1
        │   │   │   │   │   │           ├── (
        │   │   │   │   │   │           ├── <param-list>
        │   │   │   │   │   │           │   └── <id-list>
        │   │   │   │   │   │           │       ├── <id> -> table index: 9
        │   │   │   │   │   │           │       └── <id-list-cont>
        │   │   │   │   │   │           ├── )
        │   │   │   │   │   │           ├── {
        │   │   │   │   │   │           ├── <var-def-list>
        │   │   │   │   │   │           ├── <stmt-list>
        │   │   │   │   │   │           │   ├── <stmt-list>
        │   │   │   │   │   │           │   └── <stmt>
        │   │   │   │   │   │           │       └── <stmt-assign>
        │   │   │   │   │   │           │           ├── <id> -> table index: 0
        │   │   │   │   │   │           │           ├── =
        │   │   │   │   │   │           │           ├── <expr>
        │   │   │   │   │   │           │           │   └── <expr-or>
        │   │   │   │   │   │           │           │       └── <expr-and>
        │   │   │   │   │   │           │           │           └── <expr-comp>
        │   │   │   │   │   │           │           │               └── <expr-rel>
        │   │   │   │   │   │           │           │                   └── <expr-add>
        │   │   │   │   │   │           │           │                       └── <expr-mul>
        │   │   │   │   │   │           │           │                           └── <expr-unary>
        │   │   │   │   │   │           │           │                               └── <expr-primary>
        │   │   │   │   │   │           │           │                                   └── <lit>
        │   │   │   │   │   │           │           │                                       └── <lit-int>
        │   │   │   │   │   │           │           └── ;
        │   │   │   │   │   │           └── }
        │   │   │   │   │   └── <def>
        │   │   │   │   │       └── <fun-def>
        │   │   │   │   │           ├── <id> -> table index: 3
        │   │   │   │   │           ├── (
        │   │   │   │   │           ├── <param-list>
        │   │   │   │   │           ├── )
        │   │   │   │   │           ├── {
        │   │   │   │   │           ├── <var-def-list>
        │   │   │   │   │           ├── <stmt-list>
        │   │   │   │   │           │   ├── <stmt-list>
        │   │   │   │   │           │   │   ├── <stmt-list>
        │   │   │   │   │           │   │   └── <stmt>
        │   │   │   │   │           │   │       └── <stmt-assign>
        │   │   │   │   │           │   │           ├── <id> -> table index: 6
        │   │   │   │   │           │   │           ├── =
        │   │   │   │   │           │   │           ├── <expr>
        │   │   │   │   │           │   │           │   └── <expr-or>
        │   │   │   │   │           │   │           │       └── <expr-and>
        │   │   │   │   │           │   │           │           └── <expr-comp>
        │   │   │   │   │           │   │           │               └── <expr-rel>
        │   │   │   │   │           │   │           │                   └── <expr-add>
        │   │   │   │   │           │   │           │                       └── <expr-mul>
        │   │   │   │   │           │   │           │                           └── <expr-unary>
        │   │   │   │   │           │   │           │                               └── <expr-primary>
        │   │   │   │   │           │   │           │                                   └── <lit>
        │   │   │   │   │           │   │           │                                       └── <lit-int>
        │   │   │   │   │           │   │           └── ;
        │   │   │   │   │           │   └── <stmt>
        │   │   │   │   │           │       └── <stmt-fun-call>
        │   │   │   │   │           │           ├── <fun-call>
        │   │   │   │   │           │           │   ├── <id> -> table index: 7
        │   │   │   │   │           │           │   ├── (
        │   │   │   │   │           │           │   ├── <expr-list>
        │   │   │   │   │           │           │   └── )
        │   │   │   │   │           │           └── ;
        │   │   │   │   │           └── }
        │   │   │   │   └── <def>
        │   │   │   │       └── <var-def>
        │   │   │   │           ├── var
        │   │   │   │           ├── <var-list>
        │   │   │   │           │   └── <id-list>
        │   │   │   │           │       ├── <id> -> table index: 4
        │   │   │   │           │       └── <id-list-cont>
        │   │   │   │           └── ;
        │   │   │   └── <def>
        │   │   │       └── <fun-def>
        │   │   │           ├── <id> -> table index: 3
        │   │   │           ├── (
        │   │   │           ├── <param-list>
        │   │   │           ├── )
        │   │   │           ├── {
        │   │   │           ├── <var-def-list>
        │   │   │           │   ├── <var-def-list>
        │   │   │           │   └── <var-def>
        │   │   │           │       ├── var
        │   │   │           │       ├── <var-list>
        │   │   │           │       │   └── <id-list>
        │   │   │           │       │       ├── <id> -> table index: 10
        │   │   │           │       │       └── <id-list-cont>
        │   │   │           │       │           ├── ,
        │   │   │           │       │           ├── <id> -> table index: 10
        │   │   │           │       │           └── <id-list-cont>
        │   │   │           │       └── ;
        │   │   │           ├── <stmt-list>
        │   │   │           └── }
        │   │   └── <def>
        │   │       └── <var-def>
        │   │           ├── var
        │   │           ├── <var-list>
        │   │           │   └── <id-list>
        │   │           │       ├── <id> -> table index: 6
        │   │           │       └── <id-list-cont>
        │   │           └── ;
        │   └── <def>
        │       └── <fun-def>
        │           ├── <id> -> table index: 7
        │           ├── (
        │           ├── <param-list>
        │           ├── )
        │           ├── {
        │           ├── <var-def-list>
        │           │   ├── <var-def-list>
        │           │   └── <var-def>
        │           │       ├── var
        │           │       ├── <var-list>
        │           │       │   └── <id-list>
        │           │       │       ├── <id> -> table index: 11
        │           │       │       └── <id-list-cont>
        │           │       └── ;
        │           ├── <stmt-list>
        │           └── }
        └── <def>
            └── <var-def>
                ├── var
                ├── <var-list>
                │   └── <id-list>
                │       ├── <id> -> table index: 6
                │       └── <id-list-cont>
                └── ;